securedrop/qa_loader.py

```diff
@@ -0,0 +1,223 @@
+from models import (Journalist, Source, Submission, SourceStar, Reply,
+                    JournalistLoginAttempt)
+from sdconfig import config as sdconfig
+
+random.seed('~(=^–^)')  # mrow?
+
+JOURNALIST_COUNT = 10
+SOURCE_COUNT = 50
+
+
+def random_bool():
+    return bool(random.getrandbits(1))
+
+
+def random_chars(len, nullable, chars=string.printable):
+    if nullable and random_bool():
+        return None
+    else:
+        return ''.join([random.choice(chars) for _ in range(len)])
+
+
+def bool_or_none():
+    return random.choice([True, False, None])
+
+
+def random_datetime(nullable):
+    if nullable and random_bool():
+        return None
+    else:
+        return datetime(
+            year=random.randint(1, 9999),
+            month=random.randint(1, 12),
+            day=random.randint(1, 28),
+            hour=random.randint(0, 23),
+            minute=random.randint(0, 59),
+            second=random.randint(0, 59),
+            microsecond=random.randint(0, 1000),
+        )
+
+
+def new_journalist():
+    # Make a diceware-like password
+    pw = ' '.join([random_chars(3, nullable=False) for _ in range(7)])
+    journalist = Journalist(random_chars(random.randint(3, 32),
+                                         nullable=False),
+                            pw,
+                            random_bool())
+    journalist.is_admin = bool_or_none()
+
+    journalist.is_totp = bool_or_none()
+    journalist.hotp_counter = (random.randint(-1000, 1000)
+                               if random_bool() else None)
+    journalist.created_on = random_datetime(nullable=True)
+    journalist.last_access = random_datetime(nullable=True)
+
+    db.session.add(journalist)
+
+
+def new_source():
+    fid_len = random.randint(0, 32)
+    designation_len = random.randint(4, 32)
+    source = Source(random_chars(fid_len, nullable=True,
+                                 chars=string.ascii_lowercase),
+                    random_chars(designation_len, nullable=False))
+    source.flagged = bool_or_none()
+    source.last_updated = random_datetime(nullable=True)
+
+    db.session.add(source)
+
+
+def new_submission(config, source_id):
+    source = Source.query.get(source_id)
+
+    # A source may have a null fid according to the DB, but this will
+    # break storage.path.
+    if source.filesystem_id is None:
+        return
+
+    filename = fake_file(config, source.filesystem_id)
+    submission = Submission(source, filename)
+
+    # For issue #1189
+    if random_bool():
+        submission.source_id = None
+
+    submission.downloaded = bool_or_none()
+
+    db.session.add(submission)
+
+
+def fake_file(config, source_fid):
+    source_dir = path.join(config.STORE_DIR, source_fid)
+    if not path.exists(source_dir):
+        os.mkdir(source_dir)
+
+    filename = random_chars(20, nullable=False, chars=string.ascii_lowercase)
+    filename = '1-' + filename + '-msg.gpg'
+    f_len = random.randint(1, 1024*1024*500)
```

Comment:
I see why you're doing this, but sadly this produces a bunch of very large files and my test VMs ran out of space very rapidly :innocent:. 

What about reducing the number of files that have very large sizes? e.g. by drawing from a steep exponential distribution `int(math.floor(random.expovariate(10.0) * 1024 * 1024 * 500))`? We'll get mostly smaller files but (with low probability) a few bigger ones thrown in - this should significantly reduce disk space needed and still be a realistic test (note: I haven't tested this yet and it likely needs tweaking)